client/my-sites/email/inbox/new-mailbox-upsell/index.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useSelector } from 'react-redux';
+import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import { recordInboxNewMailboxUpsellClickEvent } from 'calypso/my-sites/email/email-management/home/utils';
+import { INBOX_SOURCE } from 'calypso/my-sites/email/inbox/constants';
+import { emailManagement, emailManagementEdit } from 'calypso/my-sites/email/paths';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+/**
+ * Import styles
+ */
+import './style.scss';
+
+const NewMailboxUpsell = ( { domains }: { domains: ResponseDomain[] } ) => {
+	const translate = useTranslate();
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteSlug = selectedSite?.slug;
+
+	// By default, upsell CTA links to the email management landing page.
+	let upsellURL = emailManagement( selectedSiteSlug, null, null, { source: INBOX_SOURCE } );
+
+	// User has one single domain, determine / email addition page URL based on subscribed email provider.
+	if ( domains.length === 1 ) {
+		const domainItem = domains[ 0 ];
+
+		let slug = '';
+		if ( hasTitanMailWithUs( domains[ 0 ] ) ) {
```

Comment:
I missed this one previously, but since we have the `domainItem` variable here, we should probably take the opportunity to change this to `hasTitanMailWithUs( domainItem )` (same thing for `hasGSuiteWithUs`)